misk-crypto/src/main/kotlin/misk/crypto/pgp/RealPgpDecrypter.kt

```diff
@@ -12,6 +12,14 @@ import org.bouncycastle.openpgp.jcajce.JcaPGPObjectFactory
 import org.bouncycastle.openpgp.operator.jcajce.JcePublicKeyDataDecryptorFactoryBuilder
 import org.bouncycastle.util.io.Streams
 
+/**
+ * RealPgpDecrypter is a simple PGP decryption helper.
+ * The implementation makes a  couple key assumptions:
+ * - The input data is public key encrypted.
+ * - The input data contains only ONE encrypted payload.
```

Comment:
I'm not sure this is true. We do however assume that _we_ are the only recipient currently.

Specifically we would need to modify this to search through each until we find data encrypted via a key we have the private key for:
```    val pgpPublicKeyEncryptedData =
        pgpEncryptedDataList.encryptedDataObjects.next() as PGPPublicKeyEncryptedData

    val pgpSecretKey = privateKeys[pgpPublicKeyEncryptedData.keyID]
        ?: error("no private key able to decrypt this message")```